core/src/com/unciv/models/ruleset/Building.kt

```diff
@@ -345,22 +343,34 @@ class Building : NamedStats(), IConstruction, ICivilopediaText {
         return productionCost.toInt()
     }
 
-    override fun getGoldCost(civInfo: CivilizationInfo): Int {
-        // https://forums.civfanatics.com/threads/rush-buying-formula.393892/
-        var cost = (30 * getProductionCost(civInfo)).toDouble().pow(0.75) * (1 + hurryCostModifier / 100f)
+    override fun getStatBuyCost(cityInfo: CityInfo, stat: Stat): Int? {
+        var cost = getBaseBuyCost(cityInfo, stat)
+        if (cost == null) return null
 
-        for (unique in civInfo.getMatchingUniques("Cost of purchasing items in cities reduced by []%"))
-            cost *= 1 - (unique.params[0].toFloat() / 100)
+        // Deprecated since 3.15.15
+            if (stat == Stat.Gold) {
+                for (unique in cityInfo.getMatchingUniques("Cost of purchasing items in cities reduced by []%"))
+                    cost *= 1 - (unique.params[0].toFloat() / 100)
 
-        for (unique in civInfo.getMatchingUniques("Cost of purchasing [] buildings reduced by []%")) {
-            if (isStatRelated(Stat.valueOf(unique.params[0])))
-                cost *= 1 - (unique.params[1].toFloat() / 100)
+                for (unique in cityInfo.getMatchingUniques("Cost of purchasing [] buildings reduced by []%")) {
+                    if (matchesFilter(unique.params[0]))
+                        cost *= 1 - (unique.params[1].toFloat() / 100)
+                }
+            }
+        //
+
+        for (unique in cityInfo.getMatchingUniques("[] cost of purchasing items in cities []%"))
+            if (stat.name == unique.params[0])
+                cost *= 1 + (unique.params[1].toFloat() / 100)
+
+        for (unique in cityInfo.getMatchingUniques("[] cost of purchasing [] buildings []%")) {
+            if (stat.name == unique.params[0] && matchesFilter(unique.params[1]))
+                cost *= 1 + (unique.params[2].toFloat() / 100)
         }
 
-        return (cost / 10).toInt() * 10
+        return (cost / 10f).toInt() * 10
```

Comment:
No need, cost is already Float

Reply:
I have a slight preference of doing float arithmetic with float constants, even though the generated bytecode will probably be the same. If you want I'll reverse this

Reply:
me too - due to too much SQL, where knowing where implicit conversions will tip your query plan over some edge by not allowing some statistic to be applied and thus provoking wrong cardinality assumptions is a very useful skill. This would here translate to "if you know that will be `Float.div` then feed it nothing but `Float`'s"...

Matter of taste really, I'd tend to "think SQL" in new code but leave old implicit conversions alone if it shortens the diff...